# Hypothesis Testing Question 3

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

|                |     East    |     West    |     North    |     South    |
|----------------|-------------|-------------|--------------|--------------|
|     Males      |     50      |     142     |     131      |     70       |
|     Females    |     550     |     351     |     480      |     350      |




### Steps to follow:
1. Define H0 and H1 in the question.
2. Make dimensional array.
3. Use chi square test to determine if there is a relationship between femle male buyer ratios. 
4. Define the result based on the p-value.

### Hypothesis Definition
H0 : All proportions are equal. 

H1 : Not all proportions are equal.

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as ss

In [2]:
Q3= pd.read_csv('C:\\Users\\Vignesh R Babu\\Documents\\ExcelRPython\\Assignment Codes\\3\\BuyerRatio.csv')
Q3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

### Making a dimensional Array

In [3]:
arr = np.array([[50,142,131,70],[435,1523,1356,750]]) #Dimensional Array
arr

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

### Using chi square test:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html

In [4]:
ss.chi2_contingency(arr) # OP is in the format : chi2,p-value,dof,expected

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

### Result
p-value = 0.6603 > 0.05. Therefore, we accept the Null Hypotheis that all proportions are equal.i.e. Independence of categorical variables. Thus, male-female buyer rations are similar across regions and are not related